backend/api/urls.py

```diff
@@ -0,0 +1,12 @@
+from django.urls import path
+from rest_framework.urlpatterns import format_suffix_patterns
+from api import views
+
+urlpatterns = [
+    path('', views.CoursePlanList.as_view()),
+    path('user', views.UserAPIView.as_view()),
+    path('login', views.LoginAPIView.as_view()),
+    # offset is string, because it can has negative value, which does not wok in django, so it is casted in view
+    path('weekType/<str:offset>', views.UserPlanWeekType.as_view())
```

Comment:
You can create `negint`:
https://stackoverflow.com/questions/48867977/django-2-url-path-matching-negative-value
or use re_path for that "string":
`re_path(r'^weekType/(?P<offset>[0-9-]+[0-9]), views.UserPlanWeekType.as_view())`